src/main.rs

```diff
@@ -0,0 +1,53 @@
+
+
+
+use std::fs::{self, File};
+use std::time::Instant;
+use ndarray::{Array1, Array4, Axis};
+use ndarray_npy::NpzReader;
+use rust_deep_learning::configuration::config::Config;
+use rust_deep_learning::model::sequantial::SequentialModel;
+use rust_deep_learning::Vector;
+
+fn argmax(array: &Vector) -> Option<usize> {
+    let mut max_index = None;
+    let mut max_value = std::f32::NEG_INFINITY;
+
+    for (index, &value) in array.indexed_iter() {
+        if value > max_value {
+            max_value = value;
+            max_index = Some(index);
+        }
+    }
+
+    max_index
+}
+
+fn main() {
+    let hdf5_file = hdf5::File::open("model.weights.h5").expect("Failed to open HDF5 file");
+    let deserialized: Config = serde_json::from_str(fs::read_to_string("config.json").unwrap().as_str()).unwrap();
+    let model = SequentialModel::from_config_and_hdf5(deserialized, &hdf5_file);
+    let mut npz = NpzReader::new(File::open("test_data.npz").unwrap()).unwrap();
+    let x: Array4<f32> = npz.by_name("X.npy").unwrap();
+    let y: Array1<i64> = npz.by_name("y.npy").unwrap();
```

Comment:
Consider using result and ? operator instead of unwrap

Reply:
fixed. I also add custom Error :)